app/budget/services/budgetExcelService.js

```diff
@@ -0,0 +1,456 @@
+		// Cleans data for the next sheet
+		data.length = 0;
+
+
+// SHEDULE COST REPORT
+		// Header
+		data.push(['Term',
+								'Subject Code',
+								'Course Number',
+								'Title',
+								'Units High',
+								'Units Low',
+								'Sequence',
+								'Enrollment',
+								'Sections',
+								'Instructor',
+								'Regular Instructor',
+								'Reason',								
+								'TAs',
+								'Readers',
+								'TA Cost',
+								'Reader Cost',
+								'Support Cost',
+								'Instructor Cost',
+								'Total Cost']);
+
+		viewState.calculatedScheduleCosts.terms.forEach(function(term) {
+			var scheduleCosts = viewState.calculatedScheduleCosts.byTerm[term];
+			for (var i = 0; i < scheduleCosts.length; i++) {
+				var row = [];
+				let termDescription = termDescriptions[term];
+				row.push(termDescription);
+				row.push(scheduleCosts[i].subjectCode);
+				row.push(scheduleCosts[i].courseNumber);
+				row.push(scheduleCosts[i].title);
+				row.push(scheduleCosts[i].unitsHigh);
+				row.push(scheduleCosts[i].unitsLow);
+				
+				let sectionGroupCosts = scheduleCosts[i].sectionGroupCosts;
+				for (var _i = 0; _i < sectionGroupCosts.length; _i++) {
+					let childRow = [];
+					childRow.push(sectionGroupCosts[_i].sequencePattern);
+					childRow.push(sectionGroupCosts[_i].enrollment);
+					childRow.push(sectionGroupCosts[_i].sectionCount);
+					childRow.push(sectionGroupCosts[_i].instructorDescription);
+					childRow.push(sectionGroupCosts[_i].originalInstructorDescription);
+					childRow.push(sectionGroupCosts[_i].reason);
+					childRow.push(sectionGroupCosts[_i].taCount);
+					childRow.push(sectionGroupCosts[_i].readerCount);
+					childRow.push(sectionGroupCosts[_i].taCost);
+					childRow.push(sectionGroupCosts[_i].readerCost);
+					childRow.push(sectionGroupCosts[_i].courseCostSubTotal);
+					childRow.push(sectionGroupCosts[_i].instructorCostSubTotal);
+					childRow.push(sectionGroupCosts[_i].totalCost);
+					var parentRow = row.concat(childRow);
+					data.push(parentRow);
+				}
+			}
+		});
+		
+		// Creates Costs worksheet
+		var ws_costs = XLSX.utils.aoa_to_sheet(data); // eslint-disable-line no-undef
+
+		// Set column widths
+		var wscols = [
+			{wch: 15},
+			{wch: 15},
+			{wch: 15},
+			{wch: 30},
+			{wch: 10},
+			{wch: 10},
+			{wch: 10},
+			{wch: 10},
+			{wch: 10},
+			{wch: 25},
+			{wch: 25},
+			{wch: 20},
+			{wch: 10},
+			{wch: 10},
+			{wch: 13},
+			{wch: 13},
+			{wch: 13},
+			{wch: 13},
+			{wch: 13},
+		];
+		ws_costs['!cols'] = wscols;
+
+		/* Add Costs worksheet to workbook */
+		XLSX.utils.book_append_sheet(wb, ws_costs, 'Schedule Costs'); // eslint-disable-line no-undef
+		// Cleans data for the next sheet
+		data.length = 0;
+	
+
+// FUNDS REPORT
+		var lineItems = viewState.calculatedLineItems;
+		// Header
+		data.push(['Type', 'Description', 'Amount']);
+								
+		var row = [];	
+		lineItems.forEach(function(lineItem) {
```

Comment:
Please refactor to follow code style conventions, our current code style has been to put a new line before for loops to separate them.